bin/n2log-unscramble

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/python3
+import argparse
+from datetime import datetime, timezone
+from functools import reduce
+import json
+import operator
+import re
+import sys
+
+NON_MDC = {'message', 'level', 'logger_name', 'message', 'thread_name', 'version', 'revision', 'tags', 'level_name', '@timestamp', 'stack_trace'}
+LEVELS = {
+    'trace': 0,
+    'debug': 1,
+    'info': 2,
+    'warn': 3,
+    'error': 4
+}
+
+
+def local_time(ts):
+    if ts is None:
+        return 'UNKNOWN TIME'
+
+    ts = datetime.strptime(ts, '%Y-%m-%dT%H:%M:%S.%f+00:00')
+    ts = ts.replace(tzinfo=timezone.utc)
+    ts = ts.astimezone()
+    return ts.strftime('%Y-%m-%d %H:%M:%S')
+
+
+def mdc(msg, output):
+    mdc = [(k, v) for k, v in msg.items() if k not in NON_MDC]
+    mdc.sort()
+    if mdc:
+        print()
+        for key, value in mdc:
+            print('{} = {}'.format(key, value), file=output)
+
+
+def is_level_enabled(level, level_enabled):
+    if level is None:
+        return True
+
+    if LEVELS[level_enabled] <= LEVELS[level.lower()]:
+        return True
+    return False
+
+
+def unscramble(file, output, *, level='INFO', predicate=lambda s: True):
+    for line in file:
+        try:
+            msg = json.loads(line)
+
+            if not is_level_enabled(msg.get('level'), level):
+                continue
+
+            for value in msg.values():
+                if predicate(str(value)):
+                    break
+            else:
```

Comment:
wrong line indent on line 59. `else` belongs to `if` on line 57 i guess

Reply:
No, that really does belong to the `for`.

https://docs.python.org/3/reference/compound_stmts.html#for